liana-gui/src/export.rs

```diff
@@ -343,42 +345,63 @@ impl State {
     }
 }
 
-pub async fn export_subscription(mut state: State) -> (ExportProgress, State) {
-    match state.state() {
-        Status::Init => {
-            state.start().await;
-        }
-        Status::Stopped => {
-            sleep(time::Duration::from_millis(1000)).await;
-            return (ExportProgress::None, state);
-        }
-        Status::Running => { /* continue */ }
-    }
-    let msg = state.receiver.try_recv();
-    let disconnected = match msg {
-        Ok(m) => return (m, state),
-        Err(e) => match e {
-            std::sync::mpsc::TryRecvError::Empty => false,
-            std::sync::mpsc::TryRecvError::Disconnected => true,
-        },
-    };
-
-    let handle = match state.handle.take() {
-        Some(h) => h,
-        None => return (ExportProgress::Error(Error::HandleLost), state),
-    };
-    {
-        let h = handle.lock().expect("should not fail");
-        if h.is_finished() {
-            return (ExportProgress::Finished, state);
-        } else if disconnected {
-            return (ExportProgress::Error(Error::ChannelLost), state);
+pub fn export_subscription(
+    daemon: Arc<dyn Daemon + Sync + Send>,
+    path: PathBuf,
+) -> impl Stream<Item = ExportProgress> {
+    iced::stream::channel(1, move |mut output| async move {
```

Comment:
```suggestion
    iced::stream::channel(5, move |mut output| async move {
```
We need at lesat a size of 2 for this channel but i'd choose 5 to be "conservative"